etna/models/base.py

```diff
@@ -467,40 +467,60 @@ def get_model(self) -> Any:
         pass
 
 
-class DeepBaseAbstractModel(ABC):
+class DeepAbstractNet(ABC):
     """Interface for etna native deep models."""
 
     @abstractmethod
-    def forecast(self, ts: TSDataset, horizon: int) -> TSDataset:
-        """Make predictions.
+    def make_samples(self, df: pd.DataFrame) -> Union[Iterator[dict], Iterable[dict]]:
```

Comment:
Same thing with `Iterator` and `Iterable`:

> Signature of make_samples looks strange. As I remember Iterator is Iterable because in general its method iter returns self. So if we need Iterable mentioning Iterator probably isn't necessary.